src/main/java/org/lwes/emitter/EmitterGroupBuilder.java

```diff
@@ -153,71 +139,138 @@ public static EmitterGroup createGroup(Properties props, String groupName) throw
    * @param config
    * @return
    */
   private static int getEmitCount(String config) {
-	if (STRATEGY_ALL.equalsIgnoreCase(config)) {
-	  return -1;
-	} else {
-	  Matcher mOfN = STRATEGY_M_OF_N.matcher(config);
-	  if (!mOfN.matches()) {
-		throw new IllegalArgumentException("Unable to parse nested strategy config " + config);
-	  }
-	  return Integer.parseInt(mOfN.group(1));
+  if (STRATEGY_ALL.equalsIgnoreCase(config)) {
+    return -1;
+  } else {
+    Matcher mOfN = STRATEGY_M_OF_N.matcher(config);
+    if (!mOfN.matches()) {
+    throw new IllegalArgumentException("Unable to parse nested strategy config " + config);
+    }
+    return Integer.parseInt(mOfN.group(1));
     }
   }
-  
-  private static PreserializedUnicastEventEmitter[] createEmitters(String hostsStr, int defaultPort) throws IOException {
-	String[] hosts = hostsStr.split(",");
-	PreserializedUnicastEventEmitter[] emitters = new PreserializedUnicastEventEmitter[hosts.length];
-	for (int i = 0; i < hosts.length; i++) {
-	  String host = hosts[i];
-	  int port = defaultPort;
-	  if (host.indexOf(":") > 0) {
+
+  private static DatagramSocketEventEmitter<?>[] createEmitters(String groupName, String prefix, String hostsStr, int defaultPort, EventFactory factory) throws IOException {
+    String[] hosts = hostsStr.split(",");
+    DatagramSocketEventEmitter<?>[] emitters = new DatagramSocketEventEmitter<?>[hosts.length];
+
+    for (int i = 0; i < hosts.length; i++) {
+      String host = hosts[i];
+      String ifaceStr = null;
+      Integer port = null;
+      String ttlStr = null;
+
+      // accepted formats:
+      // HOST, IFACE:HOST, HOST:PORT, IFACE:HOST:PORT, HOST:PORT:TTL, IFACE:HOST:PORT:TTL
+      if (host.indexOf(":") > 0) {
         String[] parts = host.split(":");
-		host = parts[0];
-		port = Integer.parseInt(parts[1]);
-	  } else if (defaultPort < 0) {
-		throw new RuntimeException(String.format("Unable to get port information for LWES emitter hosts: %s ",
-				hostsStr));
-	  }
-	  emitters[i] = new PreserializedUnicastEventEmitter();
-	  emitters[i].setAddress(InetAddress.getByName(host));
-	  emitters[i].setPort(port);
-	  emitters[i].initialize();
-	}
-	return emitters;
-  }  
+        if (parts.length == 2) {
+          try {
+            port = Integer.parseInt(parts[1]);
+            host = parts[0];
+          } catch (NumberFormatException nfe) {
+            ifaceStr = parts[0];
+            host = parts[1];
+          }
+        } else if (parts.length == 3) {
+          try {
+            port = Integer.parseInt(parts[1]);
+            host = parts[0];
+            ttlStr = parts[2];
+          } catch (NumberFormatException nfe) {
+            ifaceStr = parts[0];
+            host = parts[1];
+            port = Integer.parseInt(parts[2]);
+          }
+        } else if (parts.length == 4) {
+          ifaceStr = parts[0];
+          host = parts[1];
+          port = Integer.parseInt(parts[2]);
```

Comment:
This parse is outside of a try-catch.  Did we want to throw a NFE out of createEmitters()?